components/fuse/component.go

```diff
@@ -3,133 +3,242 @@ package fuse
+	}
+
+	cctx, ccancel := context.WithCancel(ctx)
+	c := &component{
+		ctx:         cctx,
+		cancel:      ccancel,
+		eventBucket: eventBucket,
+
+		congestedPercentAgainstThreshold:     congestedPercentAgainstThreshold,
+		maxBackgroundPercentAgainstThreshold: maxBackgroundPercentAgainstThreshold,
 	}
 
-	return []components.State{
-		{
-			Name:    fuse_id.Name,
-			Healthy: true,
-		},
-	}, nil
+	return c, nil
+}
+
+func (c *component) Name() string { return Name }
+
+func (c *component) Start() error {
+	go func() {
+		ticker := time.NewTicker(time.Minute)
+		defer ticker.Stop()
+
+		for {
+			c.CheckOnce()
+
+			select {
+			case <-c.ctx.Done():
+				return
+			case <-ticker.C:
+			}
+		}
+	}()
+	return nil
+}
+
+func (c *component) States(ctx context.Context) ([]components.State, error) {
+	c.lastMu.RLock()
+	lastData := c.lastData
+	c.lastMu.RUnlock()
+	return lastData.getStates()
 }
 
 func (c *component) Events(ctx context.Context, since time.Time) ([]components.Event, error) {
 	return c.eventBucket.Get(ctx, since)
 }
 
-func (c *component) Metrics(ctx context.Context, since time.Time) ([]components.Metric, error) {
-	log.Logger.Debugw("querying metrics", "since", since)
+func (c *component) Close() error {
+	log.Logger.Debugw("closing component")
+
+	c.cancel()
 
-	congestedPercents, err := metrics.ReadConnectionsCongestedPercents(ctx, since)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read congested percents: %w", err)
+	c.eventBucket.Close()
+
+	return nil
+}
+
+// CheckOnce checks the current pods
+// run this periodically
+func (c *component) CheckOnce() {
+	log.Logger.Infow("checking fuse")
+	d := Data{
+		ts: time.Now().UTC(),
 	}
-	maxBackgroundPercents, err := metrics.ReadConnectionsMaxBackgroundPercents(ctx, since)
+	defer func() {
+		c.lastMu.Lock()
+		c.lastData = &d
+		c.lastMu.Unlock()
+	}()
+
+	infos, err := fuse.ListConnections()
 	if err != nil {
-		return nil, fmt.Errorf("failed to read max background percents: %w", err)
+		d.err = err
+		return
 	}
-	ms := make([]components.Metric, 0, len(congestedPercents)+len(maxBackgroundPercents))
-	for _, m := range congestedPercents {
-		ms = append(ms, components.Metric{Metric: m})
-	}
-	for _, m := range maxBackgroundPercents {
-		ms = append(ms, components.Metric{Metric: m})
+
+	now := time.Now().UTC()
+
+	foundDev := make(map[string]fuse.ConnectionInfo)
+	for _, info := range infos {
+		// to dedup fuse connection stats by device name
+		if _, ok := foundDev[info.DeviceName]; ok {
+			continue
+		}
+
+		connsCongestedPct.WithLabelValues(Name, info.DeviceName).Set(info.CongestedPercent)
+		connsMaxBackgroundPct.WithLabelValues(Name, info.DeviceName).Set(info.MaxBackgroundPercent)
+
+		msgs := []string{}
+		if info.CongestedPercent > c.congestedPercentAgainstThreshold {
+			msgs = append(msgs, fmt.Sprintf("congested percent against threshold %.2f exceeds threshold %.2f", info.CongestedPercent, c.congestedPercentAgainstThreshold))
```

Comment:
the message is a bit weird.   should be `congested percent %.2f%% exceeds threshold %.2f%%` ?